inc/RequestParser.hpp

```diff
@@ -23,30 +24,45 @@ class RequestParser {
 public:
 	RequestParser();
 
-	void parseHttpRequest(const std::string& requestString, HTTPRequest& request);
+	void parseHeader(const std::string& headerString, HTTPRequest& request);
+	void parseBody(const std::string& bodyString, HTTPRequest& request);
 	static void clearRequest(HTTPRequest& request);
 	void clearParser();
 
+	bool hasBody() const;
+	bool isChunked() const;
+
 private:
 	bool m_hasBody;
-	bool m_chunked;
+	bool m_isChunked;
 	std::istringstream m_requestStream;
 
+	// RequestLine Parsing
+	void parseRequestLine(HTTPRequest& request);
 	static std::string parseMethod(const std::string& requestLine, HTTPRequest& request);
 	static std::string parseUri(const std::string& requestLine, HTTPRequest& request);
 	static void parseUriQuery(const std::string& requestLine, int& index, HTTPRequest& request);
 	static void parseUriFragment(const std::string& requestLine, int& index, HTTPRequest& request);
 	static std::string parseVersion(const std::string& requestLine, HTTPRequest& request);
+
+	// Header Parsing
+	void parseHeaders(HTTPRequest& request);
+
+	// Body Parsing
+	void parseChunkedBody(HTTPRequest& request);
+	void parseNonChunkedBody(HTTPRequest& request);
+
+	// Checks
```

Comment:
His comment is regarding functions that return a boolean. Is a certain condition given? Yes/no. Here we have functions which perform more complex checks and which do not return anything. That's why the naming is such.

Reply:
True, let me rephrase: I think "check" implies a simple check, which would with Chris' proposal have a is/has name. Therefore to prefix a function "check" isn't possible.

I would go for a different prefix like _validate_

Right now the check functions combine a check and state change (throw as convenience). They don't need to return because they throw. I think complexity of a check doesn't change the nature of it per se. It's just harder to answer the question e.g. is this a valid header? Even if I don't think most of them perform complex checks. checkContentLength is a bit more complex, since it also does the multiple header shebang.

I think cleaner would be to have the check function return a bool (since it checks). The caller would then decide what to do with the information (most likely state change and throw if false).
This would loose convenience, so easier would be to not use prefix check

Reply:
Renamed with prefix validate.